From mark@glines.org Wed Feb 14 10:46:35 2007
From: Mark Glines <mark@glines.org>
Date: Wed, 14 Feb 2007 10:46:21 -0800
Subject: PCI: PCI-Express capabilities, "type" field
To: Tom Long Nguyen <tom.l.nguyen@intel.com>
Cc: Greg Kroah-Hartman <gregkh@suse.de>, linux-pci <linux-pci@atrey.karlin.mff.cuni.cz>
Message-ID: <45D358FD.90704@glines.org>

The PCIE spec mentions the following list of types:

0000b (0): PCI Express Endpoint
0001b (1): Legacy PCI Express Endpoint
0100b (4): Root Port
0101b (5): Switch upstream port
0110b (6): Switch downstream port
0111b (7): Express-to-PCI/PCI-X bridge
1000b (8): PCI/PCI-X-to-Express bridge
(non-Endpoint values are only valid with a Type 1 configuration register layout)

The list in the header file is missing some entries.  So, update the list.

Signed-off-by: Mark Glines <mark@glines.org>
Cc: Tom Long Nguyen <tom.l.nguyen@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pcie/portdrv_bus.c |    2 +-
 include/linux/pcieport_if.h    |    5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/pci/pcie/portdrv_bus.c
+++ gregkh-2.6/drivers/pci/pcie/portdrv_bus.c
@@ -40,7 +40,7 @@ static int pcie_port_bus_match(struct de
 		driver->id_table->vendor != pciedev->id.vendor) ||
 	       (driver->id_table->device != PCI_ANY_ID &&
 		driver->id_table->device != pciedev->id.device) ||	
-	       (driver->id_table->port_type != PCIE_ANY_PORT &&
+	       (driver->id_table->port_type != PCIE_BR_PCIE_TO_PCI &&
 		driver->id_table->port_type != pciedev->id.port_type) ||
 		driver->id_table->service_type != pciedev->id.service_type )
 		return 0;
--- gregkh-2.6.orig/include/linux/pcieport_if.h
+++ gregkh-2.6/include/linux/pcieport_if.h
@@ -10,10 +10,13 @@
 #define _PCIEPORT_IF_H_
 
 /* Port Type */
+#define PCIE_EP_DEVICE			0	/* PCI Express endpoint device */
+#define PCIE_EP_LEGACY_DEVICE		1	/* legacy PCI Express endpoint device */
 #define PCIE_RC_PORT			4	/* Root port of RC */
 #define PCIE_SW_UPSTREAM_PORT		5	/* Upstream port of Switch */
 #define PCIE_SW_DOWNSTREAM_PORT		6	/* Downstream port of Switch */
-#define PCIE_ANY_PORT			7
+#define PCIE_BR_PCIE_TO_PCI		7	/* PCI Express-to-PCI/PCI-X bridge */
+#define PCIE_BR_PCI_TO_PCIE		8	/* PCI/PCI-X-to-PCI Express bridge */
 
 /* Service Type */
 #define PCIE_PORT_SERVICE_PME		1	/* Power Management Event */
